bakery/src/scripts/check_feed.py

```diff
@@ -6,32 +6,106 @@
 import botocore
 
 
+def flatten_feed(feed_data, feed_filter, code_version):
+    ARCHIVE_BOOK_ID_KEY = "collection_id"
+    GIT_BOOK_ID_KEY = "repository_name"
+
+    def _is_archive_entry(entry):
+        return entry.get(ARCHIVE_BOOK_ID_KEY) is not None
+
+    def _is_git_entry(entry):
+        return entry.get(GIT_BOOK_ID_KEY) is not None
+
+    def _convert_archive_entry(book, version):
+        # Archive approved book items should only have a single book entry
+        return [{
+            "collection_id": book[ARCHIVE_BOOK_ID_KEY],
+            "server": book["server"],
+            "style": book["style"],
+            "uuid": book["books"][0]["uuid"],
+            "slug": book["books"][0]["slug"],
+            "version": version
+        }]
+
+    def _convert_git_entry(book, version):
+        # Git book repo can bundle multiple books
+        result = []
+        for repo_book in book["books"]:
+            result.append({
+                "repo": book[GIT_BOOK_ID_KEY],
+                "style": book["style"],
+                "uuid": repo_book["uuid"],
+                "slug": repo_book["slug"],
+                "version": version
+            })
+        return result
+
+    books_by_id = {}
+    flattened_feed = []
+
+    if feed_filter == "archive":
+        filter_function = _is_archive_entry
+        convert_function = _convert_archive_entry
+        book_id_key = ARCHIVE_BOOK_ID_KEY
+    elif feed_filter == "git":
+        filter_function = _is_git_entry
+        convert_function = _convert_git_entry
+        book_id_key = GIT_BOOK_ID_KEY
+    else:
+        # An unexpected filter value. Just return an empty list of books
+        return []
```

Comment:
Are we sure we don't want to raise an error or fail more loudly if the `feed_filter` isn't something we expect?

Reply:
We can [raise](https://github.com/openstax/output-producer-service/pull/319/commits/ae91d8c24773895926e578c8b0f32b3cedc61558) instead.